### ISSUE TITLE
Switch impl Parser for u8 from one_of to tag(Simplify debug trace)

### ISSUE THREAD

**Issue by baoyachi** [winnow-rs/winnow#427]
### Please complete the following tasks

- [X] I have searched the [discussions](https://github.com/winnow-rs/winnow/discussions)
- [X] I have searched the [open](https://github.com/winnow-rs/winnow/issues) and [rejected](https://github.com/winnow-rs/winnow/issues?q=is%3Aissue+label%3AS-wont-fix+is%3Aclosed) issues

### rust version

rustc 1.74.0 (79e9716c9 2023-11-13)

### winnow version

0.5.34

### Minimal reproducible code

```rust
use winnow::combinator::preceded;
use winnow::prelude::*;
use winnow::error::InputError;
fn main() {

    fn parser<'s, 'a>(i: &mut &'s [u8]) -> PResult<u8, InputError<&'s [u8]>> {
        preceded(b"hello", b'a').parse_next(i)
    }
    assert_eq!(parser.parse_peek(&b"helloabc"[..]), Ok((&b"bc"[..], b'a')));
}
```

### Steps to reproduce the bug with the above code

cargo run

### Actual Behaviour

u8 originally used`one_of` combinator, debugging is more verbose, and using `tag` combinator is much simpler.

### Expected Behaviour

make debug output verbose simple

**Comment by epage** [comment]
We have #418 for doing this for `char` but I guess we can keep this open to make sure we track `u8` as well.  It is a breaking change, so we have to wait for 0.6 as it requires changes to trait bounds

**Comment by baoyachi** [comment]
Yes, this is similar to #418, but there are many changes and modifications in #418. There are still some differences in the handling of char and u8, so in order to track pull requests, I still started a new issue

**Comment by epage** [comment]
Also, for me, performance is a higher priority for what the default is than `-Fdebug` so we'll need benches. It can be even or a barely noticeable drop but can't be noticeably slower.

**Comment by epage** [comment]
To be more explicit, the way to move this forward is to have the subset of what works in 0.5 split out and merged before the 0.6 work begins.  I reserve breaking releases for only the changes that need it.  For example, I'm taking care of deprecating some API changes now and only leaving the removal of deprecated APIs for 0.6.0, rather than doing it all at once.
